prediction_market_agent/agents/microchain_agent/microchain_agent.py

```diff
@@ -53,6 +88,10 @@ def build_agent_functions(
         functions.append(
             RememberPastActions(long_term_memory=long_term_memory, model=model)
         )
+
+    wxdai_functions = build_wxdai_functions(keys=keys)
+    functions.extend(wxdai_functions)
```

Comment:
Is this for testing? Assuming we don't want to always add these functions (e.g. this will affect the current deployed agents)

Reply:
How do you envision this extra functionality @evangriffiths ?
My idea was to enhance the current agent's capabilities (following e.g. what the marketplace of tools will offer). So this would be part of the default tools, as implemented.

Reply:
I thought the agent was meant to use `ContractClassConverter` by himself, not that we would import functions from all contracts that we choose.

Reply:
Well the current set of functions the general agent has are all high level functions geared around betting in prediction markets. It's not clear to me how this will be useful when added to the current agent. Maybe we need to brainstorm a bit about the use case for this feature?

Some thoughts from me:
- At the very least, think there should be a default-off flag that controls whether to add wxdai functions to the engine, (and then you could make a new `DeployableMicrochainModifiableSystemPromptAgentAbstract` subclass that enables this, and deploy it with a terraform module). But I don't think this is a great idea, as mentioned above.
- Another option could be a microchain agent that doesn't register any smart contract-calling PMAT functions (buy/sell/...) that we've written, and only register functions for each of the necessary smart contracts (`build_wxdai_functions`, `build_conditional_tokens_functions`, ...), and deploy that agent with the same system prompt + learning functions, to compare it to the other deployed general agents.
- Following Peter's suggestion (`the agent can use ContractClassConverter by himself`) this could unlock the ability to have a Voyager-type agent that you give a list of contract addresses (e.g. those needed for prediction markets), and a high level goal (e.g. "Buy 100 Yes tokens for market 0x123..."), and it tries to generate the classes, and add/remove them from a list to achieve the goal. fwiw I don't think microchain is the right framework for this, and this is probably a pretty big task in itself.

Reply:
I agree with most of your points @evangriffiths  - 

1. Adding bool flag makes tons of sense. Added.
2,3 - Agree that we don't have a great use-case for this new functionality. I will keep this in the oven for now for future use-cases (I like the Voyager suggestion). I will create an issue for this.